lib/dill/checkpoint.rb

```diff
@@ -11,7 +11,7 @@ class << self
       attr_accessor :rescuable_errors
     end
 
-    self.rescuable_errors = [StandardError]
+    self.rescuable_errors = [StandardError, Exception]
```

Comment:
Yes, it's advisable not to catch `Exception`, so I'd be inclined to agree about catching  instead. 

I'm not sure what the rationale was for not having `RSpec::Expectations::ExpectationNotMetError` subclass `StandardError`, but if we want to add it to `rescuable_errors`, we should not do it here, since I think Dill doesn't require RSpec (or shouldn't, at least). We can do something like `Dill::Checkpoint.rescuable_errors << RSpec::Expectations::ExpectationNotMetError` to add the expectation, rather than clobbering `rescuable_errors`.